File: app/src/main/java/igrek/todotree/system/files/PathBuilder.java
package igrek.todotree.system.files;

public class PathBuilder {
    String pathstr;

    public PathBuilder(String pathstr) {
        this.pathstr = pathstr;
    }

    @Override
    public String toString() {
        return pathstr;
    }

    public static String cutSlashFromBeginning(String pathstr) {
        if (pathstr.length() > 0 && pathstr.charAt(0) == '/') {
            pathstr = pathstr.substring(1);
        }
        return pathstr;
    }

    public static String cutSlashFromEnd(String pathstr) {
        if (pathstr.length() > 0 && pathstr.charAt(pathstr.length() - 1) == '/') {
            pathstr = pathstr.substring(0, pathstr.length() - 1);
        }
        return pathstr;
    }

    public PathBuilder append(String pathstr) {
        String newPathstr = cutSlashFromEnd(this.pathstr) + "/" + cutSlashFromBeginning(pathstr);
        return new PathBuilder(newPathstr);
    }
}

File: app/src/main/java/igrek/todotree/service/filesystem/PathBuilder.kt
package igrek.todotree.service.filesystem


private fun cutSlashFromEnd(pathstr: String): String {
    var mPathStr = pathstr
    while (mPathStr.isNotEmpty() && mPathStr[mPathStr.length - 1] == '/') {
        mPathStr = mPathStr.substring(0, mPathStr.length - 1)
    }
    return mPathStr
}

fun removeExtension(pathStr: String): String {
    val lastDot = cutSlashFromEnd(pathStr).lastIndexOf(".")
    val lastSlash = cutSlashFromEnd(pathStr).lastIndexOf("/")
    if (lastDot < 0) return pathStr //nie ma kropki
    return if (lastDot < lastSlash) pathStr else pathStr.substring(
        0,
        lastDot
    )
}
